tools/test.py

```diff
@@ -471,6 +475,10 @@ def UnexpectedOutput(self):
       outcome = PASS
     return not outcome in self.test.outcomes
 
+  def HasSkipped(self):
+    skip = is_skipped.search(self.output.stdout)
```

Comment:
Just a question, since I am about to touch this code too, where is the "# SKIP" output generated? Is it logged to stdout by the tests themselves?
There is also support in the test runner to do this with status files.